src/libtest/lib.rs

```diff
@@ -71,6 +71,7 @@ use std::thread;
 use std::time::{Instant, Duration};
 
 const TEST_WARN_TIMEOUT_S: u64 = 60;
+const QUIET_MODE_MAX_COLUMN: usize = 100; // insert a '\n' after 100 tests in quiet mode
```

Comment:
Out of curiosity, I thought we printed "test xxx has run longer than 60s..." and that'd flush stdout, right? Or is that not getting printed in quiet mode?

Reply:
@alexcrichton The problem is that in quiet mode all dots were in a single line. We have >2000 run-pass tests, and only 30 minutes to spend, which means every test must complete within <0.9s to not to trigger Travis.

(Alternatively, we could fix `stamp` to output the dot immediately, instead of waiting for a `\n`.)

Reply:
Oh dear that's a good point! Sounds like we should definitely keep this.

I'd also be ok changing `stamp` though if you'd prefer. I'd imagine that we could basically print the timestamp for the first byte of each line instead of for the newline character of each line.

Reply:
I thought of updating `stamp` too, but that would require total rewrite in nonblocking mode I think.

Printing the timestamp when seeing the first byte of a line doesn't work, because Travis will still need to wait for the second to the last byte of the line. So `stamp` actually needs to do something like

```rust
loop {
    let output = await!(stdin.read_as_much_as_possible());
    if output.is_eof() { break; }
    println!("{}", output.text().replace("\n", timestamp));
}
```

Reply:
Hm I'll send a PR, I don't think we'll have to do this but I'll see if I can game it out.